lib/sacastats_web/controllers/character_controller.ex

```diff
@@ -3,143 +3,72 @@ require Logger
-    query =
-      Query.new(collection: "character")
-      |> term("name.first_lower", String.downcase(name))
-      |> resolve([
-        "online_status",
-        "world",
-        "outfit(alias,id,name)"
-      ])
-      |> join(
-        Join.new(collection: "item_profile")
-        |> on("items.item_id")
-        |> to("item_id")
-        |> list(true)
-        |> show("profile_id")
-        |> inject_at("classes_list")
-      )
-      |> lang("en")
-
-    body = query_or_redirect(conn, query, name)
-
-    next_rank =
-      body
-      |> get_in(["battle_rank", "value"])
-      |> String.to_integer()
-      |> Kernel.+(1)
-
-    status = if body["online_status"] |> String.to_integer() > 0, do: "online", else: "offline"
-
-    character = %{
-      "stat_page" => "directives.html",
-      "response" => body,
-      "next_rank" => next_rank,
-      "status" => status
-    }
-
-    render(conn, "template.html", character: character)
+  def character(conn, %{"character_name" => name, "stat_type" => "sessions"}) do
+    {:ok, info} = CensusCache.get(SacaStats.CharacterCache, name)
+    sessions = Session.get_summary(name)
+
+    {:ok, status} =
+      CensusCache.get(SacaStats.OnlineStatusCache, String.to_integer(info["character_id"]))
+
+    assigns = [
+      character_info: info,
+      online_status: status,
+      stat_page: "sessions.html",
+      sessions: sessions
+    ]
+
+    render(conn, "template.html", assigns)
   end
 
-  def character(conn, %{"character_name" => name, "stat_type" => "weapons"}) do
-    query =
-      Query.new(collection: "character")
-      |> term("name.first_lower", String.downcase(name))
-      |> resolve([
-        "online_status",
-        "outfit(alias,id,name)",
-        "weapon_stat",
-        "weapon_stat_by_faction"
-      ])
-      |> lang("en")
+  def character(%Plug.Conn{} = conn, %{"character_name" => name, "stat_type" => stat_type}) do
+    with {:ok, info} <- CensusCache.get(SacaStats.CharacterCache, name),
+         {:ok, status} <- CensusCache.get(SacaStats.OnlineStatusCache, info["character_id"]) do
+      assigns = build_assigns(conn, info, status, stat_type)
+      render(conn, "template.html", assigns)
+    else
+      {:error, :not_found} ->
+        conn
+        |> put_flash(
+          :error,
+          "Could not find a character called '#{name}'. Make sure it's spelled correctly, then try again"
+        )
+        |> redirect(to: Routes.character_path(conn, :search))
 
-    body = query_or_redirect(conn, query, name)
+      {:error, e} ->
+        Logger.error("Error fetching character: #{inspect(e)}")
+
+        conn
+        |> put_flash(
+          :error,
+          "We are unable to get that character right now. Please try again soon."
+        )
+        |> redirect(to: Routes.character_path(conn, :search))
+    end
+  end
 
-    weapon_general_stats = Enum.reduce(body["stats"]["weapon_stat"], %{}, &put_weapon_stat/2)
+  defp build_assigns(conn, info, status, "weapons") do
```

Comment:
Shouldn't it be `_conn` instead of `conn` like what this lint check is saying because it's not used in the method?